packages/renderer/src/lib/preferences/PreferencesRenderingItemFormat.svelte

```diff
@@ -169,6 +171,37 @@ async function onChange(recordId: string, value: boolean | string | number): Pro
       <FileItem record={record} value={recordValue ?? ''} onChange={onChange} />
     {:else if record.enum && record.enum.length > 0}
       <EnumItem record={record} value={recordValue} onChange={onChange} />
+    {:else if record.format?.includes(',')}
+      <!--when property supports multiple formats -->
+      {@const onFormatChange = (event: { currentTarget: { value: string | undefined } }) => {
+        selectedFormat = event.currentTarget.value;
+      }}
+      <div class="flex-col grow">
+        {#each record.format.split(',') as format}
+          <input
+            id={`${record.id}.${format}`}
+            type="radio"
+            on:change={onFormatChange}
+            checked={selectedFormat === format}
+            name={`${record.id}-format`}
+            value={format}
+            class="align-middle" />
+          <label for={`${record.id}.${format}`} class="align-middle mr-2">{startCase(format)}</label>
+        {/each}
+        <div class="flex-row">
+          {#if selectedFormat === 'file' || record.format === 'folder'}
+            <FileItem
+              record={record}
+              value={prevSelectedFormat === selectedFormat ? (recordValue ?? '') : ''}
+              onChange={onChange} />
+          {:else}
+            <StringItem
+              record={record}
+              value={prevSelectedFormat === selectedFormat ? (recordValue ?? '') : ''}
+              onChange={onChange} />
+          {/if}
```

Comment:
This will support only file and string fields. It would be nice to have some recursive component, so we don't need to repeat the logic here

Reply:
I was thinking about too, but did not want to increase number of changes in this PR.

Reply:
There could be UriInput and UriInput components too, but I am using the fact that not known formats are rendered as TextInputs. Validation is done on extension side for now, but could be implemented for specific UI components.